app/javascript/src/ActiveDocs/components/ActiveDocsSpec.jsx

```diff
@@ -1,18 +1,127 @@
 // @flow
 
 import React from 'react'
-
 import { createReactWrapper } from 'utilities/createReactWrapper'
 import SwaggerUI from 'swagger-ui-react'
-
 import 'swagger-ui-react/swagger-ui.css'
 
-type ActiveDocsSpecProps = {
+const AUTOCOMPLETE_PARAMS_MAP = {
+  'app_key': 'app_keys',
+  'app_id': 'app_ids',
+  'application_id': 'application_ids',
+  'user_key': 'user_keys',
+  'user_id': 'user_ids',
+  'account_id': 'account_ids',
+  'metric_name': 'metric_names',
+  'metric_id': 'metric_ids',
+  'backend_api_metric_name': 'backend_api_metric_names',
+  'service_id': 'service_ids',
+  'admin_id': 'admin_ids',
+  'service_plan_id': 'service_plan_ids',
+  'application_plan_id': 'application_plan_ids',
+  'account_plan_id': 'account_plan_ids',
+  'client_id': 'client_ids',
+  'client_secret': 'client_secrets',
+  'service_token': 'service_tokens',
+  'access_token': 'access_token'
+}
+
+type Primitive = string | number | boolean
+
+type AccountData = {
+  [string]: Array<{name: string, value: string}> | []
+}
+
+type SwaggerResponse = {
+  body: {
+    [string]: mixed
+  },
+  data: string,
+  headers: {},
+  obj: {},
+  ok: boolean,
+  status: number,
+  statusText: string,
+  text: string,
   url: string
 }
 
-const ActiveDocsSpec = ({ url }: ActiveDocsSpecProps) => <SwaggerUI url={url} />
+type ActiveDocsSpecProps = {
+  accountData: AccountData | {},
+  url: string,
+}
+
+type ParamArraySchema = {
+  type: 'array',
+  items: { type: Primitive }
+}
+type ParamEnumSchema = {
+  enum: Array<Primitive> | [],
```

Comment:
Here I don't think you need the second part, just 
```suggestion
  enum: Array<Primitive>,
```
does the trick